lib/spring/generate/springInitializrGenerator.ts

```diff
@@ -14,26 +14,44 @@
  * limitations under the License.
  */
 
+import { RemoteLocator } from "@atomist/automation-client/operations/common/params/RemoteLocator";
+import { RepoCreationParameters } from "@atomist/automation-client/operations/generate/RepoCreationParameters";
+import {
+    SmartParameters,
+    ValidationResult,
+} from "@atomist/automation-client/SmartParameters";
 import { doWithRetry } from "@atomist/automation-client/util/retry";
-import { NodeFsLocalProject, ParametersObject, Project, SoftwareDeliveryMachine } from "@atomist/sdm";
+import {
+    NodeFsLocalProject,
+    Parameter,
+    Parameters,
+    Project,
+    SoftwareDeliveryMachine,
+} from "@atomist/sdm";
 import axios from "axios";
 import * as decompress from "decompress";
 import * as fs from "fs";
 import * as path from "path";
 import * as tmp from "tmp-promise";
-import { SetAtomistTeamInApplicationYml } from "./springBootTransforms";
 import {
-    SpringProjectCreationParameterDefinitions,
-    SpringProjectCreationParameters,
-} from "./SpringProjectCreationParameters";
+    JavaIdentifierRegExp,
+    JavaPackageRegExp,
+    MavenArtifactIdRegExp,
+    MavenGroupIdRegExp,
+} from "../../java/javaPatterns";
+import { SetAtomistTeamInApplicationYml } from "./springBootTransforms";
+import { SpringProjectCreationParameters } from "./SpringProjectCreationParameters";
 import { TransformSeedToCustomProject } from "./transformSeedToCustomProject";
 
-export function addSpringInitializrGenerator(sdm: SoftwareDeliveryMachine) {
+const springBootMetaData: any = {};
+
+export async function addSpringInitializrGenerator(sdm: SoftwareDeliveryMachine) {
+    await getSpringInitializrMetaData().then(response => springBootMetaData.metaData = response);
     sdm.addGeneratorCommand<SpringInitializrProjectCreationParameters>({
         name: "start.spring.io",
         intent: "spring initializr",
         description: "Create a new Spring Boot project using Spring Initializr",
-        parameters: SpringInitializrProjectCreationParameterDefinitions,
+        paramsMaker: () => new SpringInitializrProjectCreationParameters(),
```

Comment:
Could you make a single argument constructor for `SpringInitializrProjectCreationParameters` that sets a metadata property on the object to avoid the "global" springBootMetaData?